File: backend/app/services/email_sender.py
import logging
import smtplib
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText

from app.core.config import settings

logger = logging.getLogger(__name__)


def send_gmail(
    to_email: str, subject: str, content: str, sender_name: str = ""
) -> dict:
    if not settings.smtp_user or not settings.smtp_password:
        logger.warning("SMTP not configured — returning simulated send")
        return {"simulated": True, "message": "SMTP not configured"}

    name = sender_name.strip() or "CVDrop-AI"
    msg = MIMEMultipart()
    msg["From"] = f"{name} <{settings.smtp_user}>"
    msg["To"] = to_email
    msg["Subject"] = subject
    msg.attach(MIMEText(content, "plain", "utf-8"))

    try:
        server = smtplib.SMTP(settings.smtp_host, settings.smtp_port)
        server.starttls()
        server.login(settings.smtp_user, settings.smtp_password)
        server.sendmail(settings.smtp_user, to_email, msg.as_string())
        server.quit()
        logger.info("Email sent to %s", to_email)
        return {"simulated": False, "message": "Email sent successfully"}
    except Exception as exc:
        logger.error("Failed to send email: %s", exc)
        return {"simulated": True, "message": str(exc)}

File: frontend/lib/email-sender.ts
import nodemailer from "nodemailer";

export function sendGmail(
  toEmail: string,
  subject: string,
  content: string,
  senderName: string = ""
): { simulated: boolean; message: string } {
  const smtpUser = process.env.SMTP_USER || "";
  const smtpPassword = process.env.SMTP_PASSWORD || "";

  if (!smtpUser || !smtpPassword) {
    console.warn("SMTP not configured — returning simulated send");
    return { simulated: true, message: "SMTP not configured" };
  }

  const name = senderName.trim() || "CVDrop-AI";
  const transporter = nodemailer.createTransport({
    host: process.env.SMTP_HOST || "smtp.gmail.com",
    port: Number(process.env.SMTP_PORT) || 587,
    secure: false,
    auth: { user: smtpUser, pass: smtpPassword },
  });

  try {
    transporter.sendMail({
      from: `"${name}" <${smtpUser}>`,
      to: toEmail,
      subject,
      text: content,
    });
    console.info("Email sent to %s", toEmail);
    return { simulated: false, message: "Email sent successfully" };
  } catch (exc) {
    console.error("Failed to send email:", exc);
    return { simulated: true, message: String(exc) };
  }
}
